Signal/src/Models/OWSMessagesBubblesSizeCalculator.m

```diff
@@ -76,6 +89,46 @@ - (CGSize)messageBubbleSizeForMessageData:(id<JSQMessageData>)messageData
     }
 }
 
+- (CGSize)sizeForSystemMessage:(TSInteraction *)interaction
+                      cacheKey:(id)cacheKey
+                        layout:(JSQMessagesCollectionViewFlowLayout *)layout
+{
+    OWSAssert(interaction);
+    OWSAssert(cacheKey);
+
+    NSValue *cachedSize = [self.cache objectForKey:cacheKey];
+    if (cachedSize != nil) {
+        return [cachedSize CGSizeValue];
+    }
+
+    CGSize result =
+        [OWSSystemMessageCell cellSizeForInteraction:interaction collectionViewWidth:layout.collectionView.width];
```

Comment:
The sizing logic has been moved into the cells for the systems messages and the unread indicator, to gather concerns and to leverage the constants defined there.